server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/HAProxyMessageHandler.java

```diff
@@ -60,6 +69,7 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                 // Refresh MDC info to account for proxying
                 MDCBuilder boundMDC = IMAPMDCContext.boundMDC(ctx);
 
+                connectionChecks.forEach(connectionCheck -> Mono.from(connectionCheck.validate((InetSocketAddress) ctx.channel().remoteAddress())).block());
```

Comment:
IMO we should get the client IP from `HAProxyMessage`, not parsing again.

Maybe `haproxyMsg.sourceAddress()` is the client IP instead? 

TBH I am not sure. Please test more: at least try docker-compose with haproxy and try to see which one is the client IP. The current IMAPServerTest can not be fully trusted IMO.

After being sure about the client IP, this work should be good IMO.

Reply:
Yep, you're right, `haproxyMsg.sourceAddress()` is the client IP 
I will adapt the code later. Thank for your comments!!
![image](https://github.com/apache/james-project/assets/87489927/d0922550-5d80-4575-b201-221158235171)

![image](https://github.com/apache/james-project/assets/87489927/444c8ed0-c678-4f1c-8f13-a191c75c4f3c)

Reply:
Flux.fromIterable(...).concatMap(...)>then().block();

Please use reactor control structures.

Reply:
> I will adapt the code later.

Now is about the time.